Splay Tree/SplayTree.swift

```diff
@@ -0,0 +1,369 @@
+/*
+  A splay tree is a self-adjusting binary search tree with the additional
+  property that recently accessed elements are quick to access again.
+ 
+  It performs basic operations such as insertion, look-up and removal in O(log n)
+  amortized time.
+ 
+  For many sequences of non-random operations, splay trees perform better than
+  other search trees, even when the specific pattern of the sequence is unknown.
+ */
+public class SplayTree<T: Comparable> {
+  
+  private var root: SplayTreeNode<T>?
+  
+  //MARK: Public
+  
+  public convenience init(_ values: [T]) {
+    
+    self.init()
+    
+    for value in values {
+      insert(value: value)
+    }
+  }
+  
+  func insert(value: T) {
+    
+    var current: SplayTreeNode<T>? = root
+    var parent: SplayTreeNode<T>?
+    
+    while (current != nil) {
```

Comment:
I think it would be nice to make this let binding, so you don't have to constantly force unwrap every single line with **current**.